### DISCUSSION TITLE
How to mutate values inside a record?

### DISCUSSION THREAD

**Discussion by dedebenui** [nushell/nushell#12521] [Q&A]
If I want to add some path to my `PATH` environment, I can do
```nushell
$env.PATH ++= [some_path]
```

Now say I have the following record:
```nushell
mut df = {users: [[name, groups, email]; [John, [admin, www], null], [Alice, [root, admin, teachers], alice@example.com]], location: Outside, start: (date now)}
```

I want to append a few groups to the user `John`. My intuition is to do 

```nushell
$df.users.0.groups ++= [Hello world]
```

I get no error but `$df` remains unchanged. I get an error when using `let df = ...`

Is that a bug? What's another syntax I can use?

edit:
here is my `version` info

| key                | value                               |
| ------------------ | ----------------------------------- |
| version            | 0.92.2                              |
| branch             |                                     |
| commit_hash        |                                     |
| build_os           | macos-aarch64                       |
| build_target       | aarch64-apple-darwin                |
| rust_version       | rustc 1.77.2 (25ef9e3d8 2024-04-09) |
| rust_channel       | stable-aarch64-apple-darwin         |
| cargo_version      | cargo 1.77.2 (e52e36006 2024-03-26) |
| build_time         | 2024-04-12 15:51:23 +02:00          |
| build_rust_channel | release                             |
| allocator          | mimalloc                            |
| features           | default, sqlite, trash, which       |
| installed_plugins  | msgpack, plist                      |

**Comment by ghost** [comment]
im currently still on 0.91 and for me your example works (the `| table -e` has updated).  
maybe a regression / bug

another more verbose method for doing this (which might work for you):
```nu
$df = ($df | update users.0.groups {|i| $i.users.0.groups | append [Hello world]})
```

**Comment by dedebenui** [comment]
Good catch. I've submitted an issue (#12532)
